src/seraphis_main/enote_record_utils_legacy.cpp

```diff
@@ -366,6 +367,197 @@ bool try_get_legacy_basic_enote_record(const LegacyEnoteVariant &enote,
+static bool is_legacy_enote_v1(const cryptonote::transaction &tx, const cryptonote::tx_out &out)
+{
+    // Plaintext amount, no view tag
+    return (tx.version == 1 || (tx.version == 2 && cryptonote::is_coinbase(tx))) &&
+        out.target.type() == typeid(cryptonote::txout_to_key);
+}
+//-------------------------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------------------------
+static bool is_legacy_enote_v2(const cryptonote::transaction &tx, const cryptonote::tx_out &out)
+{
+    // Encrypted amount v1, no view tag
+    return tx.version == 2 && !cryptonote::is_coinbase(tx) &&
+        out.target.type() == typeid(cryptonote::txout_to_key) && is_encoded_amount_v1(tx);
+}
+//-------------------------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------------------------
+static bool is_legacy_enote_v3(const cryptonote::transaction &tx, const cryptonote::tx_out &out)
+{
+    // Encrypted amount v2, no view tag
+    return tx.version == 2 && !cryptonote::is_coinbase(tx) &&
+        out.target.type() == typeid(cryptonote::txout_to_key) && is_encoded_amount_v2(tx);
+}
+//-------------------------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------------------------
+static bool is_legacy_enote_v4(const cryptonote::transaction &tx, const cryptonote::tx_out &out)
+{
+    // Plaintext amount, view tag
+    return (tx.version == 1 || (tx.version == 2 && cryptonote::is_coinbase(tx))) &&
+        out.target.type() == typeid(cryptonote::txout_to_tagged_key);
+}
+//-------------------------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------------------------
+static bool is_legacy_enote_v5(const cryptonote::transaction &tx, const cryptonote::tx_out &out)
+{
+    // Encrypted amount v2, view tag
+    return tx.version == 2 && !cryptonote::is_coinbase(tx) &&
+        out.target.type() == typeid(cryptonote::txout_to_tagged_key) && is_encoded_amount_v2(tx);
+}
+//-------------------------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------------------------
+static bool try_out_to_legacy_enote_v1(const cryptonote::transaction &tx,
+    const size_t output_index,
+    sp::LegacyEnoteVariant &enote_out)
+{
+    if (output_index >= tx.vout.size())
+        return false;
```

Comment:
This is a minor performance change, but if you change the function signature of `try_out_to_legacy_enote_vx` to have a reference `const cryptonote::tx_out&` (like `is_legacy_enote_vx`) instead of an output index, then you won't have to check this `if` branch 5 times for every single enote.

Reply:
The reason I didn't do that is because the amount commitments and encrypted amounts aren't on the `tx_out`, they're in `tx.rct_signatures.outPk[output_index]` and `tx.rct_signatures.ecdhInfo[output_index]` respectively, and passing those in by reference and handling each `try_out_to_legacy` function differently felt clunky

Reply:
Fair enough... it doesn't matter much either way